controllers/storagecluster/external_resources.go

```diff
@@ -308,7 +308,7 @@ func (r *StorageClusterReconciler) createExternalStorageClusterResources(instanc
 				scc = newCephFilesystemStorageClassConfiguration(instance)
 				enableRookCSICephFS = true
 			} else if d.Name == cephRbdStorageClassName {
-				scc = newCephBlockPoolStorageClassConfiguration(instance)
+				scc = newCephBlockPoolStorageClassConfiguration(instance, d.Data["dataPool"])
```

Comment:
Don't pass `d.Data["dataPool"]`, better to just pass `d.Data`, in case we need something else in the future. Then in `newCephBlockPoolStorageClassConfiguration()` determine whether `d.Data["dataPool"]` exists or not.

Reply:
now, passing `d.Data` only.